modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStore.java

```diff
@@ -232,8 +230,6 @@ private boolean read0(
             // If page was not written yet, nothing to read.
             if (n < 0) {
                 pageBuf.put(new byte[pageBuf.remaining()]);
```

Comment:
Is this even possible? Maybe we should remove this part as well. What's the point of returning an empty page.
I have a hypothesis that this is part of the code of partition initialization from Ignite 2.x that's also obsolete

Reply:
I tried to analyze the code associated with this place and it seems that there may be a moment when we want to write the page to disk, but we only wrote it to a temporary buffer for writing it, and another thread wants to read it.

I would not touch this place yet, maybe come back to it later, wdyt?

Reply:
Maybe later, ok.
>we want to write the page to disk, but we only wrote it to a temporary buffer for writing it, and another thread wants to read it.

Other thread should read from buffer then. We need a page content, not an empty page in this case, right? Looks kinda buggy to me.

Reply:
Let's investigate it later, ok? Please create a separate JIRA issue.

Reply:
Created IGNITE-17397